pyro/infer/enum.py

```diff
@@ -33,12 +33,14 @@ def iter_discrete_extend(trace, site, **ignored):
         yield extended_trace
 
 
-def get_importance_trace(graph_type, max_plate_nesting, model, guide, *args, **kwargs):
+def get_importance_trace(graph_type, max_plate_nesting, model, guide, *args, detach=False, **kwargs):
```

Comment:
@neerajprad this interface is becoming awkward. WDYT of replacing `*args, **kwargs` with `args, kwargs` in a later cleanup PR, either before or after this merges?

Reply:
I think it makes sense to only use this `*model_args, **model_kwargs` pattern for user-facing interface, and even then, its debatable if there are too many other arguments. I'll put up a fix.